src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -93,7 +94,8 @@ g.test('dynamic set to true is allowed only for buffers')
   .params(poptions('type', kBindingTypes))
   .fn(async t => {
     const { type } = t.params;
-    const success = kBindingTypeInfo[type].type === 'buffer';
+    const success =
+      kPerPipelineBindingLimits[kBindingTypeInfo[type].perPipelineBindingLimitType].maxDynamic > 0;
```

Comment:
Would we be able to remove this two-level indirection? Or perhaps expose the binding limits on the BindingTypeInfo object so that you don't have to know to import another table to get them and instead do `kBindingTypeInfo[type].pipelineBindingLimits.maxDynamic`.

Perhaps we can have `kBindingTypeInfo = expand(kBindingTypeInfo)` where
```typescript
function expand(info: BindingTypeInfo): ExpandedThing {
  return {
      ...info,
      pipelineBindingLimits: kPerPipelineBindingLimits[info.perPipelineBindingLimitType]
  };
}
```

Reply:
I was thinking about this but I'm not sure it makes sense. maxDynamic is not a property of the binding type because it is shared across multiple binding types. However "supportsDynamic" could be a property of the binding type...

Reply:
maybe just behind a function on the object, so it's not directly part of it? I think it'll be a bit nicer to use than looking up in another table. Even this would be nice:

```typescript
return {
      ...info,
      getPipelineBindingLimits: () => kPerPipelineBindingLimits[info.perPipelineBindingLimitType]
};
```

Reply:
Eh, probably just removing the indirection would be ok. Probably not that unclear.